pkg/closenicely/closeutil.go

```diff
@@ -0,0 +1,13 @@
+package closenicely
+
+import "go.uber.org/zap"
+
+type CloserWithError interface {
+	Close() error
+}
+
+func OrDebug(closer CloserWithError) {
+	if err := closer.Close(); err != nil {
+		zap.S().Debug("Failed to close resource: %s", err)
```

Comment:
Maybe pass the error as a field so the Logger can choose how to format (eg, with stacktrace)?